registry/remote/repository.go

```diff
@@ -437,6 +437,59 @@ func filterReferrers(refs []artifactspec.Descriptor, artifactType string) []arti
 	return refs[:j]
 }
 
+// Extension provides information on extensions provided on a Registry.
+// Reference: https://github.com/opencontainers/distribution-spec/blob/main/extensions/_oci.md#extensions-property-descriptions
+type Extension struct {
+	Name        string   `json:"name"`
+	URL         string   `json:"url"`
+	Description *string  `json:"description,omitempty"`
+	Endpoints   []string `json:"endpoints"`
+}
```

Comment:
The go struct from the spec may be reused.

Reference: https://github.com/opencontainers/distribution-spec/blob/main/specs-go/v1/extensions/ext.go